consumer/src/mode/decoded/deposited_curve.rs

```diff
@@ -0,0 +1,333 @@
+            .block_timestamp(event.block_timestamp)
+            .transaction_hash(event.transaction_hash.clone())
+            .build();
+
+        event
+            .upsert(
+                &decoded_consumer_context.pg_pool,
+                &decoded_consumer_context.backend_schema,
+            )
+            .await
+            .map_err(ConsumerError::ModelError)
+    }
+
+    /// This function initializes accounts for sender and receiver
+    async fn initialize_accounts(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+    ) -> Result<(), ConsumerError> {
+        // Create or get the sender account
+        get_or_create_account(
+            self.sender.to_string(),
+            decoded_consumer_context,
+        )
+        .await?;
+
+        // Create or get the receiver account
+        get_or_create_account(
+            self.receiver.to_string(),
+            decoded_consumer_context,
+        )
+        .await?;
+
+        Ok(())
+    }
+
+    /// This function handles the creation of a curve deposit
+    pub async fn handle_curve_deposit_creation(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+        event: &DecodedMessage,
+    ) -> Result<(), ConsumerError> {
+        info!("Processing DepositedCurve event: {:?}", self);
+
+        // Initialize accounts
+        self.initialize_accounts(decoded_consumer_context).await?;
+
+        // Get or create the curve vault
+        let curve_vault = self.get_or_create_curve_vault(decoded_consumer_context, event).await?;
+
+        // Create deposit record first
+        let deposit_id = self.create_deposit(decoded_consumer_context, event).await?;
+
+        // Create event with deposit_id
+        self.create_event(event, decoded_consumer_context, &deposit_id).await?;
+
+        // Handle position
+        self.handle_position(decoded_consumer_context, &curve_vault).await?;
+
+        // Create signal
+        self.create_signal(decoded_consumer_context, event, &curve_vault).await?;
+
+        Ok(())
+    }
+
+    /// This function gets or creates a curve vault
+    async fn get_or_create_curve_vault(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+        event: &DecodedMessage,
+    ) -> Result<CurveVault, ConsumerError> {
+        // Get the current share price
+        let current_share_price = decoded_consumer_context
+            .fetch_current_share_price(self.vaultId, event.block_number)
+            .await?;
+
+        // Get the base vault to determine if this is for an atom or triple
+        let base_vault = Vault::find_by_id(
+            U256Wrapper::from(self.vaultId),
+            &decoded_consumer_context.pg_pool,
+            &decoded_consumer_context.backend_schema,
+        )
+        .await?
+        .ok_or(ConsumerError::VaultNotFound)?;
+
+        // Use the curveId from the event as the curve number
+        let curve_number = U256Wrapper::from(self.curveId);
```

Comment:
We can do away with vault_id entirely, and just use atom_id, triple_id, counter_triple_id.  For curves, we have an additional curve_id (I think called curve_number in this PR at the moment).

Queries would look like querying for `atomId` and getting back all vaults including curves, or querying for `atomId` and `curveId` to get a specific curve vault only.